bundles/org.openhab.binding.sonnen/src/main/java/org/openhab/binding/sonnen/internal/communication/SonnenJSONCommunication.java

```diff
@@ -38,21 +39,64 @@ public class SonnenJSONCommunication {
 
     private Gson gson;
     private @Nullable SonnenJsonDataDTO batteryData;
+    private SonnenJsonPowerMeterDataDTO @Nullable [] powerMeterData;
 
     public SonnenJSONCommunication() {
         gson = new Gson();
         config = new SonnenConfiguration();
     }
 
     /**
-     * Refreshes the battery connection.
+     * Refreshes the battery connection with the new API Call V2.
      *
      * @return an empty string if no error occurred, the error message otherwise.
      */
-    public String refreshBatteryConnection() {
+    public String refreshBatteryConnectionAPICALLV2() {
         String result = "";
-        String urlStr = "http://" + config.hostIP + "/api/v1/status";
+        String urlStr = "http://" + config.hostIP + "/api/v2/status";
+        Properties httpHeader = new Properties();
+        httpHeader = createHeader(config.authToken);
+        try {
+            String response = HttpUtil.executeUrl("GET", urlStr, httpHeader, null, "application/json", 10000);
+            logger.debug("BatteryData = {}", response);
+            if (response == null) {
+                throw new IOException("HttpUtil.executeUrl returned null");
+            }
+            batteryData = gson.fromJson(response, SonnenJsonDataDTO.class);
+
+            if (config.powerMeter) {
```

Comment:
Is this configuration needed? Instead, perhaps you could check if any of the four new channels are linked.

Reply:
Actually I think it is cleaner to do this way. The reaso is that by initializing the binding in the MainUI of openhab there is a option in advanced which need to be activated by the users if he wants to have the PowerMeter Data, as the standard user will use it. Then I query that specific API call only when the user activated it otherwise not.

Reply:
With my proposal you would also query the API only when needed. The disadvantages I see in the current approach:
- The user needs to make this additional decision, which also needs to be documented.
- The decision is a technical decision connected to the implementation details of the API. This should be abstracted.
- It's possible to set the configuration to `true` without linking any of the channels. In this case the API calls are wasted.
- It's possible to set the configuration to `false` and link some of the channels. The user may then fail to understand why the channels are not updated.

Reply:
I have thought about it. "Unfortunately" 👍  you are right. Therefore I changed the code and commited the newest version. I hope it fits now.